### ISSUE TITLE
no documentation for re-exported renamed modules

### ISSUE THREAD

**Issue by ghc-mirror** [haskell/haddock#225]
**Original reporter**: _claude@<hidden>_

With a deep module heirarchy and modules re-exporting many modules, I find this syntax much less verbose:

``` haskell
module Foo (module M) where
import Bar as M
import Baz as M
```

But haddock makes no useful documentation.

The verbose repetitive syntax works as expected with links to the re-exported modules:

``` haskell
module Foo
  ( module Bar
  , module Baz
  ) where
import Bar
import Baz
```

It would be very nice if haddock recognised module renaming and generated appropriate links to the re-exported modules - as it is now I'm going to have to go back to listing everything twice so that documentation is useable.

**Comment by tfausak** [comment]
Duplicate of #121?

**Comment by ntc2** [comment]
Issue #584 also mentions this issue among others.

**Comment by gbaz** [comment]
should be resolved by #688
